packages/core/src/combo-box/ComboBox.tsx

```diff
@@ -342,6 +343,7 @@ export const ComboBox = forwardRef(function ComboBox<Item>(
     event.stopPropagation();
     const removed = selectedState[index];
     removePill(event, removed);
+    onPillRemove?.(event, index);
```

Comment:
@tayyabataimur can we explain the original user's issue with this?
We now have 2 apis which do the same thing onSelectionChange and onPillRemove, one is value driven and one is index driven ?
Also, it's only called when the pill is removed and not if I deselect in the list, so there is a danger of 2 apis returning different values ?

onPillRemove is part of Pill's API, it's bled into the ComboBox API, but probably should of been removed from the ComboBox API to avoid this confusion?

Any thoughts?

Reply:
ie... onPillRemove is only called when you interact with the pill but another API onSelectionChange is called when the list changes.